topnav/media-menu.js

```diff
@@ -0,0 +1,127 @@
+
+  static get properties() {
+    return {
+      mediaMenuOpen: { type: Boolean },
+      mediaMenuAnimate: { type: Boolean },
+      mediaSliderOpen: { type: Boolean },
+      mediaSliderAnimate: { type: Boolean }
+    };
+  }
+
+  constructor() {
+    super();
+    this.mediaSliderOpen = false;
+    this.mediaSliderAnimate = false;
+  }
+
+  mediaSlider() {
+    // Only keep side menu open until menu closed
+    // Shift menu downwards as per button clicked
+    this.mediaSliderAnimate = true;
+    this.mediaSliderOpen = !this.mediaSliderOpen;
+  }
+
+  render() {
+    const mediaMenuClass = this.mediaMenuOpen ? 'media-menu open slide-in' : this.mediaMenuAnimate ? 'media-menu slide-out' : 'media-menu';
+    const mediaMenuHidden = this.mediaMenuOpen ? 'false' : 'true';
+    const mediaMenuExpanded = this.mediaMenuOpen ? 'true' : 'false';
+    return html`
+    <nav
+      class="${mediaMenuClass}"
+      aria-hidden="${mediaMenuHidden}"
+      aria-expanded="${mediaMenuExpanded}"
+    >
+      <!-- Include icon and name inline in a button-->
+      <div class="grid">
+        <button tabindex="-1" @click="${this.mediaSlider}"><div><img src="assets/img/ia-waybackmachine-999.svg"></div></button>
+        <button><div>Wayback Machine</div></button>
+        <button tabindex="-1" @click="${this.mediaSlider}"><div><img src="assets/img/ia-texts-999.svg"></div></button>
+        <button><div>Texts</div></button>
+        <button tabindex="-1" @click="${this.mediaSlider}"><div><img src="assets/img/ia-video-999.svg"></div></button>
+        <button><div>Video</div></button>
+        <button tabindex="-1" @click="${this.mediaSlider}"><div><img src="assets/img/ia-audio-999.svg"></div></button>
+        <button><div>Audio</div></button>
+        <button tabindex="-1" @click="${this.mediaSlider}"><div><img src="assets/img/ia-software-999.svg"></div></button>
+        <button><div>Software</div></button>
+        <button tabindex="-1" @click="${this.mediaSlider}"><div><img src="assets/img/ia-images-999.svg"></div></button>
+        <button><div>Images</div></button>
+        <button tabindex="-1" @click="${this.mediaSlider}"><div><img src="assets/img/ia-more-999.svg"></div></button>
+        <button><div>More</div></button>
+      </div>
+    </nav>
+    <media-slider ?mediaSliderOpen="${this.mediaSliderOpen}" ?mediaSliderAnimate="${this.mediaSliderAnimate}"></media-slider>
+    `;
+  }
+
+  static get styles() {
+    return css`
+      .media-menu {
+        width: 100%;
+        height: 500px;
+        background-color: var(--black);
+        color: var(--white);
+        margin: 0px;
+        font-size: 20px;
+        font-family: "Helvetica Neue";
+        transform: translate(0px, -1000px);
+      }
+      .media-menu button {
+        position: relative;
+        background: none;
+        color: inherit;
+        border: none;
+        font: inherit;
+        cursor: pointer;
+        text-align: left;
+        z-index: 5;
```

Comment:
Question: why 5 as z-index? can we use a lower number, like 2? since you use 1 below.

Reply:
The earlier z indexes are already being used by the other components.
z indexes alotted according to the order of appearance:
z-index 1: User menu
z-index 2: Topnav
z-index 3: Search menu
z-index 4: Media menu
z-index 5: Media slider

Reply:
Question: can we use DOM element order to minimize the number of z-indexes used?

Reply:
Most of the z indexes are required to keep track of the overlapping of menus and animations since we have multiple cases of the same. For example, while the user menu animation is supposed to pop out from behind the topnav, the search menu is supposed to pop out on top of it. And in all cases the media menu is supposed to overlap the others while being animated.